snapcraft/tests/fixture_setup.py

```diff
@@ -875,3 +877,57 @@ def fetch_binary(self, dir_, progress):
 
     def get_dependencies(self, _):
         return []
+
+
+class WithSnapNotInstalled(fixtures.Fixture):
+
+    def __init__(self, snap_name):
+        super().__init__()
+        self.snap_name = snap_name.split('/')[0]
+
+    def setUp(self):
+        super().setUp()
+        if snapcraft.repo.snaps.SnapPackage.is_snap_installed(self.snap_name):
+            raise AssertionError(
+                "This test cannot run if you already have the {snap!r} snap "
+                "installed. Please uninstall it by running "
+                "'sudo snap remove {snap}'.".format(snap=self.snap_name))
+
+        self.addCleanup(self._remove_snap)
+
+    def _remove_snap(self):
+        try:
+            subprocess.check_output(
+                ['sudo', 'snap', 'remove', self.snap_name],
+                stderr=subprocess.STDOUT)
+        except subprocess.CalledProcessError as e:
+            RuntimeError("unable to remove {!r}: {}".format(
+                self.snap_name, e.output))
+
+
+class SnapcraftYaml(fixtures.Fixture):
+
+    def __init__(
+            self, path, name='test-snap', version='test-version',
+            summary='test-summary', description='test-description'):
+        super().__init__()
+        self.path = path
+        self.data = {
+            'name': name,
+            'version': version,
+            'summary': summary,
+            'description': description,
+        }
+
+    def add_part(self, name, data):
```

Comment:
Add a `*` after `self`.

Reply:
@sergiusens I'm trying to understand the rule we are following for named arguments.
`add_part('my-part', {plugin: 'nil'}` seems clear to me.

`add_part(name='my-part', data={plugin: 'nil'}` doesn't add much, IMO. I'm not against adding it, not a big issue to me. I just want to understand when I should and when I shouldn't.